src/main/java/picard/vcf/FixVcfHeader.java

```diff
@@ -121,41 +114,54 @@ public static void main(final String[] args) {
         if (HEADER != null) IOUtil.assertFileIsReadable(HEADER);
         IOUtil.assertFileIsWritable(OUTPUT);
 
-        final VCFFileReader reader = new VCFFileReader(INPUT, false);
+        final VCFFileReader reader     = new VCFFileReader(INPUT, false);
+        final VCFHeader existingHeader = reader.getFileHeader();
 
         final VCFHeader outHeader;
         if (HEADER != null) { // read the header from the file
             final VCFFileReader headerReader = new VCFFileReader(HEADER, false);
+            final VCFHeader inputHeader      = headerReader.getFileHeader();
             if (ENFORCE_SAME_SAMPLES) {
-                outHeader = headerReader.getFileHeader();
-                enforceSameSamples(reader.getFileHeader(), outHeader);
+                outHeader = inputHeader;
+                enforceSameSamples(existingHeader, outHeader);
             }
             else {
-                outHeader = new VCFHeader(headerReader.getFileHeader().getMetaDataInInputOrder(), reader.getFileHeader().getSampleNamesInOrder());
+                outHeader = new VCFHeader(inputHeader.getMetaDataInInputOrder(), existingHeader.getSampleNamesInOrder());
             }
             CloserUtil.close(headerReader);
+            outHeader.getFilterLines()
+                    .stream()
+                    .filter(filterHeaderLine -> !existingHeader.hasFilterLine(filterHeaderLine.getID()))
+                    .forEach(filterHeaderLine -> log.info("FILTER line found in HEADER will be added to OUTPUT: " + filterHeaderLine));
             outHeader.getInfoHeaderLines()
                     .stream()
-                    .filter(infoHeaderLine -> !reader.getFileHeader().hasInfoLine(infoHeaderLine.getID()))
+                    .filter(infoHeaderLine -> !existingHeader.hasInfoLine(infoHeaderLine.getID()))
                     .forEach(infoHeaderLine -> log.info("INFO line found in HEADER will be added to OUTPUT: " + infoHeaderLine.getID()));
             outHeader.getFormatHeaderLines()
                     .stream()
-                    .filter(formatHeaderLine -> !reader.getFileHeader().hasInfoLine(formatHeaderLine.getID()))
+                    .filter(formatHeaderLine -> !existingHeader.hasInfoLine(formatHeaderLine.getID()))
                     .forEach(formatHeaderLine -> log.info("FORMAT line found in HEADER will be added to OUTPUT: " + formatHeaderLine.getID()));
         }
         else { // read the input
             final ProgressLogger progress = new ProgressLogger(log, 1000000, "read");
-            final VCFFileReader in = new VCFFileReader(INPUT, false);
-            final VCFHeader inHeader = in.getFileHeader();
+            final VCFFileReader in        = new VCFFileReader(INPUT, false);
```

Comment:
`in` is never closed

Reply:
Closed on line 176/192 below but I have changed this block to be a try-with-resources